eclipse-che/che-server
# fix: support for kubernetes cluster domain

default kuberentes cluster domain is cluster.local. however several
setups may have different cluster domains.

Signed-off-by: Kazım SARIKAYA <kazimsarikaya@sanaldiyar.com>



### What does this PR do?

The default format is defined at:

https://github.com/eclipse-che/che-server/blob/d640df4d7fdba2721a90983ac29640c0e9136273/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/TraefikGatewayRouteConfigGenerator.java#L58

We use multiple kubernetes with different cluster domain different than cluster.local. Hence this static value blocks us to use che at our environments. If this format can be a envrionment variable, it will be useful for deployments like ours.

I added the property with default

```
che.infra.kubernetes.cluster_domain = cluster.local
```

hence we can change this property with env variable 

```
CHE_INFRA_KUBERNETES_CLUSTER__DOMAIN
```

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/18927
This problem occurred an example like:

``` 
kubeadm init --service-dns-domain   "myorg.internal"
``` 

when this kind of setups ```TraefikGatewayRouteConfigGenerator``` still generates service url formats ending with **cluster.local**


### How to test this PR?


Setup kubernetes cluster with non default dns domain such as 

``` 
kubeadm init --service-dns-domain   "myorg.internal"
``` 

Setup che with gateway support (traefik)

The related ```jwtproxy``` configmaps still contains **cluster.local**  like:

``` yaml
   http:
      routers:
        serveruhsrmsdx-jwtproxy-webviews:
          rule: "PathPrefix(`/serveruhsrmsdx-jwtproxy/webviews/`)"
          service: "serveruhsrmsdx-jwtproxy-webviews"
          middlewares:
          - "serveruhsrmsdx-jwtproxy-webviews"
          priority: 100
      services:
        serveruhsrmsdx-jwtproxy-webviews:
          loadBalancer:
            servers:
            - url: "http://serveruhsrmsdx-jwtproxy.<namespace>.svc.cluster.local:<port>"
      middlewares:
        serveruhsrmsdx-jwtproxy-webviews:
          stripPrefix:
            prefixes:
            - "/serveruhsrmsdx-jwtproxy/webviews/"
```

However the url should end with **myorg.internal** 

This pr fixes this problem.


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.
